cli/cmd/report_definitions_diff.go

```diff
@@ -0,0 +1,166 @@
+//
+// Author:: Darren Murray(<darren.murray@lacework.net>)
+// Copyright:: Copyright 2023, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/pkg/errors"
+	"github.com/pmezard/go-difflib/difflib"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+// reportDefinitionsDiffCommand command is used to compare 2 lacework report definition versions
+var reportDefinitionsDiffCommand = &cobra.Command{
+	Use:   "diff <report_definition_id> <version> <version>",
+	Short: "Compare 2 versions of a report definition",
+	Long: `Compare 2 versions of a report definition.
+
+To see a diff of 2 report definition versions:
+
+    lacework report-definition diff <report_definition_id> <current_version> <new_version>
+`,
+	Args: cobra.ExactArgs(3),
+	RunE: diffReportDefinition,
+}
+
+func diffReportDefinition(_ *cobra.Command, args []string) error {
+	var (
+		err        error
+		versionOne int
+		versionTwo int
+		reportOne  *diffCfg
+		reportTwo  *diffCfg
+	)
+
+	if versionOne, err = strconv.Atoi(args[1]); err != nil {
+		return errors.Wrap(err, "unable to parse version")
+	}
+
+	if versionTwo, err = strconv.Atoi(args[2]); err != nil {
+		return errors.Wrap(err, "unable to parse version")
+	}
+
+	cli.StartProgress("Fetching all report definition versions...")
+	response, err := cli.LwApi.V2.ReportDefinitions.GetVersions(args[0])
+	cli.StopProgress()
+
+	if err != nil {
+		return err
+	}
+
+	for _, r := range response.Data {
+		if r.Version == versionOne {
+			reportOne = &diffCfg{
+				name:   fmt.Sprintf("Version-%d", r.Version),
+				object: r,
+			}
+		}
+		if r.Version == versionTwo {
+			reportTwo = &diffCfg{
+				name:   fmt.Sprintf("Version-%d", r.Version),
+				object: r,
+			}
+		}
+	}
+
+	if reportOne == nil || reportTwo == nil {
+		return errors.New("unable to find report definition versions")
+	}
+
+	diff, err := diffAsYamlString(*reportOne, *reportTwo)
```

Comment:
What if both versions are the same?

Reply:
response is "" same as a git diff if results are the same
We could prevent the user comparing the same version.